.github/workflows/promote-staging-to-prod.yml

```diff
@@ -0,0 +1,77 @@
+name: Promote Staging to Prod
+
+on:
+  workflow_dispatch: {}
+  repository_dispatch:
+    types: [deployment]
+
+jobs:
+  check_staging_commit_checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: staging
+      - name: Check if Latest Staging Post-Deploy Tests are Passing
+        id: get_checks
+        run: |
+          checks=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+                       "https://api.github.com/repos/chanzuckerberg/single-cell-data-portal/actions/runs?actor=czibuildbot" | jq -r '.workflow_runs[] | select(.display_title == "stage") | .conclusion' | head -n 1)
+          if [ "$checks" == "success" ]; then
+            echo "All checks passed!"
+          else
+            echo "Some checks failed." 
+            exit 1
+          fi
+        shell: bash
+
+  merge_staging_to_prod:
+    runs-on: ubuntu-latest
+    needs:
+      - check_staging_commit_checks
+    steps:
+      - name: Checkout prod branch
+        uses: actions/checkout@v2
+        with:
+          ref: prod
+          fetch-depth: 0 # fetch all history to make merges work correctly
+      - name: Merge staging branch into prod branch
+        run: |
+          git config user.name "GitHub Actions"
+          git config user.email "actions@github.com"
+
+          echo "Checking out 'prod' branch"
+          git checkout prod
+          echo "Confirming checked out branch receiving merge is: $(git branch --show-current)"
+
+          echo "Latest commit on 'prod' branch is: $(git log -1 --format='%H' prod)"
+          echo "Latest commit on 'staging' branch is: $(git log -1 --format='%H' staging)"
+
+          echo "About to merge 'staging' branch into 'prod' branch"
+          git merge --verbose staging -m "Merging staging branch into prod branch"
```

Comment:
this operation is what should be done in the Makefile rather than bothering to use a personal access token to use the github API to call this GHA

Reply:
I like having the GHA set-up because I like having the UI button push option, but I'll factor this out into a script that both command line + UI can use

Reply:
🙏 yeah I think having the GHA is good because the UI button feature is nice, but I would replace the `make` recipe with a regular `git`-based operation so that we don't introduce the personal access token dependency

Reply:
Resolved with added script 👍